src/orchestratord/src/controller/materialize/resources.rs

```diff
@@ -0,0 +1,1250 @@
+        args.push(format!(
+            "--aws-external-id-prefix={}",
+            &mz.spec.external_id_prefix
+        ));
+        if let Some(azs) = config.aws_info.environmentd_availability_zones.as_ref() {
+            for az in azs {
+                args.push(format!("--availability-zone={az}"));
+            }
+        }
+
+        if let Some(environmentd_connection_role_arn) = &config.aws_info.environmentd_iam_role_arn {
+            args.push(format!(
+                "--aws-connection-role-arn={}",
+                environmentd_connection_role_arn
+            ));
+        }
+        if let Some(account_id) = &config.aws_info.aws_account_id {
+            args.push(format!("--aws-account-id={account_id}"));
+        }
+
+        args.extend([
+            format!("--aws-secrets-controller-tags=Owner={}", mz.oidc_sub()),
+            format!(
+                "--aws-secrets-controller-tags=Environment={}",
+                mz.name_unchecked()
+            ),
+        ]);
+        args.extend_from_slice(&config.aws_info.aws_secrets_controller_tags);
+    }
+
+    // Add Kubernetes arguments.
+    args.extend([
+        "--orchestrator=kubernetes".into(),
+        format!(
+            "--orchestrator-kubernetes-service-account={}",
+            &mz.service_account_name()
+        ),
+        "--orchestrator-kubernetes-service-node-selector=workload=materialize-instance".into(),
+        "--orchestrator-kubernetes-image-pull-policy=if-not-present".into(),
+    ]);
+    if let Some(scheduler_name) = &config.scheduler_name {
+        args.push(format!(
+            "--orchestrator-kubernetes-scheduler-name={}",
+            scheduler_name
+        ));
+    }
+    for (key, val) in mz.default_labels() {
+        args.push(format!(
+            "--orchestrator-kubernetes-service-label={key}={val}"
+        ));
+    }
+    if let Some(image) = &config.init_container_image {
+        args.push(format!(
+            "--orchestrator-kubernetes-init-container-image={image}"
+        ));
+    }
+
+    // Add logging and tracing arguments.
+    args.extend(["--log-format=json".into()]);
+    if let Some(endpoint) = &tracing.opentelemetry_endpoint {
+        args.extend([
+            format!("--opentelemetry-endpoint={}", endpoint),
+            format!(
+                "--opentelemetry-resource=organization_id={}",
+                mz.spec.organization_id
+            ),
+            format!(
+                "--opentelemetry-resource=environment_name={}",
+                mz.name_unchecked()
+            ),
+        ]);
+    }
+
+    args.extend([
+        "--orchestrator-kubernetes-ephemeral-volume-class=openebs-lvm-instance-store-ext4"
+            .to_string(),
+        // The `materialize` user used by clusterd always has gid 999.
+        "--orchestrator-kubernetes-service-fs-group=999".to_string(),
+    ]);
+
+    // Add Sentry arguments.
+    if let Some(sentry_dsn) = &tracing.sentry_dsn {
```

Comment:
🗒️ another bit to flag for future changes, feels like we could push a lot of these args for sentry, fronteg etc into `envd_extra_args` to generify a bit